azure-iot-device/azure/iot/device/common/chainable_exception.py

```diff
@@ -14,3 +14,8 @@ def __init__(self, message=None, cause=None):
         # while still being able to operate in Python 2.
         self.__cause__ = cause
         super(ChainableException, self).__init__(message)
+
+    def __repr__(self):
+        return "{} caused by {}".format(
+            super(ChainableException, self).__repr__(), self.__cause__.__repr__()
+        )
```

Comment:
Interesting. Don't know when this would actually be used (since the `__str___` inherited from Exception would take precedence), but for any situation where `__repr__` was invoked this would be helpful.

Reply:
The default __str__ calls __repr__.  I think I did this because some of Logging() code uses __repr__ and I wanted to see this in logger.error() output.

---
In reply to: [333675697](https://github.com/Azure/azure-iot-sdk-python/pull/347#discussion_r333675697) [](ancestors = 333675697)